src/Avalonia.Controls/Converters/ProgressRingEndsToPathDataConverter.cs

```diff
@@ -0,0 +1,134 @@
+        {
+            double angle0 = (double)values[2];
+            
+            double angle1 = (double)values[3];
+
+            double startAngle;
+            double endAngle;
+
+            if (values.Count > 4) //we have min+max+value to work with
+            {
+                startAngle = NO_PROGRESS;
+
+                double min = angle0;
+                double max = angle1;
+                
+                double range = max - min;
+
+                double value = (double)values[4];
+                
+                endAngle = DegreesToRad((((value - min) / range) * 360) - 90);
+
+            }
+            else //indeterminate
+            {
+                angle0 = DegreesToRad(angle0);
+                angle1 = DegreesToRad(angle1);
+                startAngle = Math.Min(angle0, angle1);
+                endAngle = Math.Max(angle0, angle1);
+            }
+
+            double angleGap = RadToNormRad(endAngle - startAngle);
+
+            Rect trackBounds = (Rect)values[0];
+
+            double strokeInset = (double)values[1] / 2;
+
+            double trackCenterX = (trackBounds.Width / 2);
+            double trackCenterY = (trackBounds.Height / 2);
+            double trackRadiusX = trackCenterX - strokeInset;
+            double trackRadiusY = trackCenterY - strokeInset;
+
+            double normStart = RadToNormRad(startAngle);
+            double normEnd = RadToNormRad(endAngle);
+            
+            if ((normStart == normEnd) && (startAngle != endAngle))
+            {
+                return new EllipseGeometry()
+                {
+                    RadiusX = trackRadiusX,
+                    RadiusY = trackRadiusY,
+                    Center = new Point(trackCenterX, trackCenterY)
+                };
+
+                /*Point ringPoint = GetRingPoint(trackRadiusX, trackRadiusY, trackCenterX, trackCenterY, startAngle);
+                return new PathGeometry()
+                {
+                    Figures = 
+                    {
+                        new PathFigure()
+                        {
+                            StartPoint = ringPoint,
+                            Segments = 
+                            {
+                                new ArcSegment()
+                                {
+                                    Point = ringPoint,
+                                    IsLargeArc = true,
+                                    Size = new Size(trackRadiusX, trackRadiusY),
+                                    SweepDirection = SweepDirection.Clockwise
+                                }
+                            },
+                            IsClosed = true
+                        }
+                    }
+                };*/
+            }
+            else if ((normStart == normEnd) && (startAngle == endAngle))
+            {
+                return new PathGeometry();
+            }
+            else
+            {
+                return new PathGeometry()
```

Comment:
Not sure if geometry creation each animation step will have nice performance.

Reply:
Fixed. A nice side effect of this fix is that we'll also be welcoming a new control into the `Avalonia.Controls.Shapes` family: `Arc`.